### PR TITLE
metrics: dump the error metrics during failure

### PR DESCRIPTION
**Pull request by bchalios**
## Changes

Fix the assertion to print a more useful message when an error metric is non-zero. Also, make sure that we upload metrics to CloudWatch even when there is a non-zero error metric.

## Reason

Right now the assertion message is not very legible which makes it difficult to debug. Also, if an error happens we don't upload the metrics to cloudwatch.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this
  PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by roypat — APPROVED** [review]
thanks :heart:

File: tests/host_tools/fcmetrics.py [hunk 1]
```diff
@@ -463,14 +464,15 @@ def flatten_dict(node, prefix: str):
...
+
+    failure_metrics = {
         key: value
         for key, value in flattened_metrics.items()
         if "err" in key or "fail" in key or "panic" in key or "num_faults" in key
         if value
     }
-
-    metrics.flush()
+    assert not failure_metrics, json.dumps(failure_metrics)
```
Comment (pb8o) [inline tests/host_tools/fcmetrics.py:475]: would it be worth to pretty print with `indent=True`?

Comment (pb8o) [reply tests/host_tools/fcmetrics.py:475]: Also why is it not legible? Shouldn't pytest pretty print it for us?

Comment (bchalios) [reply tests/host_tools/fcmetrics.py:475]: The output currently looks like this: https://buildkite.com/firecracker/performance-nightly-metrics/builds/273#019143e5-33d0-4707-9987-f6ff987f44c9/50-239. I can do the `ident=True`  change

**Review by roypat — APPROVED** [review]

**Review by pb8o — APPROVED** [review]

**Pull request merged**